### PR TITLE
Adding IAM instance profile for elastic beanstalk

### PR DESCRIPTION
**Pull request by warrenbailey**
**What**
Adds an IAM instance profile for elastic beanstalk to speed up deployments.

**How to test**
1. Deploy terraform and check everything still works
2. Log into AWS, go to elastic beanstalks settings and change something. Notice the warning no longer exists.

**Who can review**
Anyone apart from @warrenbailey

### CODE REVIEW

**Comment by collisdigital** [comment]
Reading http://docs.aws.amazon.com/elasticbeanstalk/latest/dg/iam-instanceprofile.html

It appears EB already creates an instance profile called "aws-elasticbeanstalk-ec2-role" with the required permissions that also has an sts:AssumeRole trust relationship setup. Checking our AWS accounts this role does exist and matches the doc above. Is it possible to use the existing role? The one being added in this pull request defines a Trust Relationship but doesn't actually grant any permissions, so it's not clear if it will help (other than suppressing the EB warning).

**Comment by collisdigital** [comment]
See also: https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/concepts-roles.html

**Comment by warrenbailey** [comment]
I wasn't hugely happy with that IAM role I created so that sounds like an excellent idea. I'll investigate.

**Comment by warrenbailey** [comment]
Using the provided IAM role work, so this is now ready for review

**Comment by collisdigital** [comment]
I think the way to test this is effective would be to remove the CloudWatch access key and secret from survey runner and see that boto picks up the instance profile permissions instead and can still write to CloudWatch etc.

**Comment by warrenbailey** [comment]
All tested and it works, so I've removed the AWS access/secret keys from the survey runner deployment as it no longer needs them

File: README.md [hunk 1]
```diff
@@ -32,6 +32,10 @@ For the smoke test you will need
 2. Gem 2.5.2 (gem update --system '2.5.2')
 3. Bundle should be 1.10.6 (gem install bundler -v 1.10.6)
 
+## IAM Role
+Currently for this to work you need to add the CloudWatchFullAccess policy to the IAM role aws-elasticbeanstalk-ec2-role
```
Comment (collisdigital) [inline README.md:36]: Lets get this working in Pre-Prod and then use the access advisor to narrow down the permissions needed before moving to production.

File: global_vars.tf [hunk 2]
```diff
@@ -192,4 +192,20 @@ variable "database_user" {
 variable "database_password" {
   description = "The password for the master username of the database"
   default = "digitaleq12345"
+}
+
+variable "elastic_beanstalk_iam_role" {
+  default = "aws-elasticbeanstalk-ec2-role"
+}
+
+variable "eq_server_side_storage" {
```
Comment (collisdigital) [inline global_vars.tf:201]: Is this meant to be in this PR and below?

**Comment by weapdiv-david** [comment]
LGTM 👍

**Pull request merged**